src/ddmd/dsymbolsem.d

```diff
@@ -5403,9 +5403,12 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         buildOpAssign(sd, sc2);
         buildOpEquals(sd, sc2);
 
-        sd.xeq = buildXopEquals(sd, sc2);
-        sd.xcmp = buildXopCmp(sd, sc2);
-        sd.xhash = buildXtoHash(sd, sc2);
+        if (!global.params.betterC)  // these functions are used for TypeInfo
```

Comment:
Due to #7288, I think this should be checking the `useTypeInfo` flag.

Reply:
done

Reply:
The splitting of the betterC options was obviously coming ;), glad it's in now.

Reply:
Yes, I introduced that.  Please continue to enforce. :-)